### PR TITLE
Support for BotD Zoo: export some classes, eliminate implicit dependencies

### PR DESCRIPTION
**Pull request by xnerhu**
This pull request introduces changes necessary to support broyster in botd zoo. Mainly exposing some APIs and refactoring parts to take values explicitly instead of taking them from e.g. env file. I tried to not introduce major breaking changes.

### CODE REVIEW

File: node/src/browserstack_session_factory.ts
Comment (Finesse) [inline node/src/browserstack_session_factory.ts:1]: 👍

**Review by Finesse — CHANGES_REQUESTED** [review]

**Comment by Finesse** [comment]
> Mainly exposing some APIs and refactoring parts to take values explicitly instead of taking them from e.g. env file.

Could you please work with @fpkamp to ensure that no class or function reads the credentials from the environment variables (gets them from the arguments instead)? For Karma, please create a separate function that only reads the credentials from the environment variables, and add this function to [the dependency collection](https://github.com/fingerprintjs/broyster/blob/6e42e19d1230c0c9fd2a56a9abe8a89b8e7afed9/node/src/karma_plugin.ts#L10).

**Comment by xnerhu** [comment]
> > Mainly exposing some APIs and refactoring parts to take values explicitly instead of taking them from e.g. env file.
> 
> Could you please work with @fpkamp to ensure that no class or function reads the credentials from the environment variables (gets them from the arguments instead)? For Karma, please create a separate function that only reads the credentials from the environment variables, and add this function to [the dependency collection](https://github.com/fingerprintjs/broyster/blob/6e42e19d1230c0c9fd2a56a9abe8a89b8e7afed9/node/src/karma_plugin.ts#L10).

The previous implementation of `BrowserStackSessionFactory` loaded username and access token from `.env` file. In this PR I delegated store authentication data in `CapabilitiesFactory` which is passed to `BrowserStackSessionFactory`, but `CapFactory` and `BrowserStackSessionFactory` themselves don't load any env variables.

Regarding dependency collection, `makeBrowserStackSessionFactor` is already added. I think I shouldn't add `makeCapabilitiesFactory`

@Finesse

**Comment by Finesse** [comment]
> The previous implementation of `BrowserStackSessionFactory` loaded username and access token from `.env` file. In this PR I delegated store authentication data in `CapabilitiesFactory` which is passed to `BrowserStackSessionFactory`, but `CapFactory` and `BrowserStackSessionFactory` themselves don't load any env variables.
> 
> Regarding dependency collection, `makeBrowserStackSessionFactor` is already added. I think I shouldn't add `makeCapabilitiesFactory`

@xnerhu You made `makeBrowserStackSessionFactor` read the credentials from environment variables (they don't necessarily come from an `.env` file). I meant creating a separate Karma dependency for the credentials to fix the problem that you want to solve in all classes and functions. It is not your scope, I'll do it, never mind.

File: example_project/.env.example [hunk 1]
```diff
@@ -0,0 +1,2 @@
+BROWSERSTACK_USERNAME=
+BROWSERSTACK_ACCESS_KEY=
```
[the remark this answers is not in the document]
Comment (Finesse) [reply example_project/.env.example]: @xnerhu In our team, the person who resolves a conversation is the person who starts the conversation. It is to ensure that the notice is really solved.

File: node/src/index_node.ts [hunk 2]
```diff
@@ -2,3 +2,5 @@ export { default as karmaPlugin } from './karma_plugin'
 export { sslConfiguration } from './server_certificates'
 export * as httpHttpsServer from './http_https_server'
 export { setHttpsAndServerForKarma } from './karma_https_config'
+export * from './browserstack_session_factory'
+export * from './capabilities_factory'
```
[the remark this answers is not in the document]
Comment (Finesse) [reply node/src/index_node.ts]: Please describe the new exports in the documentation.

**Comment by Finesse** [comment]
@xnerhu What do you think about [these changes](https://github.com/fingerprintjs/broyster/pull/47/commits/b67f03edc29487cceee9dc90865ff9f80a4f663e)?

**Comment by fpkamp** [comment]
Looks good, @Finesse should we wait for the documentation?

**Comment by Finesse** [comment]
> Looks good, @Finesse should we wait for the documentation?

@fpkamp Yes. If you'd like, feel free to add the documentation.

**Review by xnerhu — COMMENTED** [review]
Looks good

File: node/readme.md [hunk 3]
```diff
@@ -17,29 +17,31 @@ import * as broysterForNode from '@fpjs-incubator/broyster/node'
...
         The HTTP server runs on the port provided by Karma, while the HTTPS port will run on +1 from that.
-    - `setHttpsAndServerForKarma` That configures karma for HTTP and HTTPS testing without any additional work.
-- `@fpjs-incubator/broyster/browser`:
-    - `retryFailedTests` That allows overriding the different behavior of Jasmine specs. The new behavior will retry a failed test up until the maximum specified in the first parameter, with a delay between each such attempt, indicated by the second parameter (in miliseconds). Call this function in the root of any executable file, involved in your testing code, for example, in a Jasmine helper file. Once called, it affects all tests Jasmine runs, even in the other files. For Karma, you can add a file that contains the invocation and point it in your `files`, that way you will not have it tied to one specific test file.
+    -   `setHttpsAndServerForKarma` That configures karma for HTTP and HTTPS testing without any additional work.
+    -   `CapabilitiesFactory` Stores credentials for BrowserStack
```
Comment (fpkamp) [inline node/readme.md:29]: The caps factory actually creates an object that describes the parameters of the driver session that will be requested

File: node/readme.md [hunk 4]
```diff
@@ -144,3 +146,24 @@ The following config options are available inside the browserStack section of th
...
 
 Currently supported flags: Incognito, Headless (case insensitive). Multiword flags will be stripped of dashes.
+
+### Using Selenium directly
+
+```js
+import { CapabilitiesFactory, BrowserStackSessionFactory } from '@fpjs-incubator/broyster'
+
+const local = false // Execute webdriver commands on BrowserStack remotely
+const capabilitiesFactory = new CapabilitiesFactory({ username, password }, local)
```
Comment (Finesse) [inline node/readme.md:160]: A parameter name is incorrect: `{ username, accessKey }`

**Review by Finesse — CHANGES_REQUESTED** [review]

**Review by Finesse — APPROVED** [review]

**Pull request merged**